### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by RiyaBulia12**
### Project Requirement
Link: https://github.com/microverseinc/curriculum-ruby/blob/main/simple-ruby/create_your_own_enumerable.md

### Steps
- Created a class `MyList` that has an instance variable `@list`.
- In `MyList` implemented a method `#each` that yields successive members of `@list` and uses the `MyEnumerable` module.
- Created a module `MyEnumerable` that implements the following methods which  has the same functionality as methods in [Enumerable](https://ruby-doc.org/core-3.0.0/Enumerable.html)):
  - `#all?`
  - `#any?`
  - `#filter`
- Each class and module should have a separate .rb file.

### Collaborators
@shaqdeff 
@RiyaBulia12 

### Result Screenshot
Screenshot 2022-11-04 at 5 13 30 PM

### CODE REVIEW

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @RiyaBulia12 and @shaqdeff  👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️ 

- [ ] Your linter checks are not passing. Please make sure they are passing to proceed 👍 
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require_relative 'enumerable'
+
+class MyList
+  include MyEnumerable
+
+  def initialize(arr)
+    @list = arr
+  end
```
Comment (AmaduKamara) [inline my_list.rb]: - [x] When I run the project with the sample test command, it returns an error indicating that, expected 1 argument and passed 3. In your initialize method, it should take a list of arguments and not only 1.

- You could have done something in this way:
```ruby
    def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

- [x] Kindly work on that 👍 

**See the error screenshot below**
image

If you run this test command below:
`ruby -r "./my_list.rb" -e "p MyList.new(1,2,4)"`

You should get this result
`<MyList: @list=[1, 2, 4]>`

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @RiyaBulia12,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter check passed 👍 
✅ Good descriptive PR 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: my_list.rb (hunk 1 above)
Comment (RiyaBulia12) [reply my_list.rb]: @AmaduKamara Thanks for pointing that out we were looking for that solution only which accepts n args instead of 1

**Review by BertrandConxy — APPROVED** [review]
Hi @RiyaBulia12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**